vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java

```diff
@@ -210,12 +231,29 @@ public void deselectAll() {
     public void updateSelection(Set<T> addedItems, Set<T> removedItems) {
         Objects.requireNonNull(addedItems, "added items cannot be null");
         Objects.requireNonNull(removedItems, "removed items cannot be null");
-        doUpdateSelection(addedItems, removedItems, false);
+
+        Map<Object, T> addedItemsMap = mapItemsById(addedItems);
+        Map<Object, T> removedItemsMap = mapItemsById(removedItems);
+        addedItemsMap.keySet().stream().filter(removedItemsMap::containsKey)
+                .collect(Collectors.toList()).forEach(key -> {
+                    addedItemsMap.remove(key);
+                    removedItemsMap.remove(key);
+                });
+        doUpdateSelection(addedItemsMap, removedItemsMap, false);
     }
 
     @Override
     public boolean isSelected(T item) {
-        return selected.containsKey(getItemId(item));
```

Comment:
The naming probably could be `isSelectedItem()` to be more consistent with method names as we have an `isSelectedItemId()` now.

Reply:
We shouldn't change existing public API, it would be a breaking change.

Reply:
Thanks for the response, it makes sense, then this will be a minor inconsistency within public API.